python/ambassador/fetch/ingress.py

```diff
@@ -105,17 +105,20 @@ def _update_status(self, obj: KubernetesObject) -> None:
             )
 
     def _resolve_service_port_number(self, namespace, service_name, service_port):
-        for k8s_svc in self.service_dep.discovered_services.values():
-            key = f"{k8s_svc.name}.{k8s_svc.namespace}"
+        self.logger.debug(f"Resolving named port '{service_port}' in service '{service_name}'")
 
-            if key != f"{service_name}.{namespace}":
-                continue
+        key = KubernetesObjectKey(KubernetesGVK("v1", "Service"), namespace, service_name)
+        k8s_svc: Optional[KubernetesObject]
+        k8s_svc = self.service_dep.discovered_services.get(key, None)
+        if not k8s_svc:
+            self.logger.debug(f"Could not find service '{service_name}'")
+            return service_port
 
-            for port in k8s_svc.spec.get("ports", []):
-                if service_port == port.get("name", None):
-                    return port.get("port", service_port)
-            self.logger.debug(f"Could not resolve port '{service_port}' for service '{key}'")
+        for port in k8s_svc.spec.get("ports", []):
+            if service_port == port.get("name", None):
+                return port.get("port", service_port)
 
+        self.logger.debug(f"Could not find port '{service_port}' in service '{service_name}'")
```

Comment:
since line 119 also could also return the service port, this debug log wouldn't always get outputted if `service_port` is returned

Reply:
Correct. If the port is found we don't need to display "not found" message